File: php/yield/list_slice.php
<?php

function list_slice($list, $n) {
	$slice_list = [];
	foreach ($list as $val) {
		$slice_list[] = $val;
		if (count($slice_list) === $n) {
			yield $slice_list;
			$slice_list = [];
		}
	}

	// padding
	if (!empty($slice_list)) {
		$i = $n - count($slice_list);
		while ($i--) {
			$slice_list[] = 0;
		}

		yield $slice_list;
	}
}

$numbers = range(1, 22);
foreach (list_slice($numbers, 4) as $slice_nums) {
	var_dump($slice_nums);
}

File: python/yield/list_slice.py
#! coding: utf-8

def list_slice(list, n):
    slice_list = []
    for i in list:
        slice_list.append(i)
        if len(slice_list) == n:
            yield slice_list
            slice_list = []

    if slice_list != []:
        i = n - len(slice_list)
        while i != 0:
            slice_list.append(0)
            i -= 1

        yield slice_list


for nums in list_slice(range(1, 22), 4):
    print nums
